lib/livereload.coffee

```diff
@@ -78,8 +78,14 @@ class Server
     exts = @config.exts
     fileext = path.extname filepath
                   .substring 1
+
     for ext in exts when ext == fileext
-      @refresh filepath
+      delayedRefresh = setTimeout(
+        =>
+          clearTimeout(delayedRefresh)
+          @refresh filepath
+        @config.delay
```

Comment:
Let me take a closer look. I was under the impression that it was on a per
extension basis rather than a per file. the 0 check makes sense, will make
a commit to address that separately before reasoning about the other issue.
Thanks.

On Wednesday, May 25, 2016, Brian Hogan <notifications@github.com> wrote:

> In lib/livereload.coffee
> <https://github.com/napcs/node-livereload/pull/67#discussion_r64582118>:
>
> >      for ext in exts when ext == fileext
> > -      @refresh filepath
> > +      delayedRefresh = setTimeout(
> > +        =>
> > +          clearTimeout(delayedRefresh)
> > +          @refresh filepath
> > +        @config.delay
>
> I have questions about this routine. Even if there is no delay specified,
> this is still going to create and drop a timeout. That seems wasteful.
> Probably should check if delay is 0?
>
> Also, this creates a timeout for each file. If I'm watching 25 files, will
> this create 25 delays? I'm shaky on how that works.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/napcs/node-livereload/pull/67/files/51ced5dec695810ad95faf056787236a222017b8..19f4efbb7855e90b15787c570f3e082a3b46c6c1#r64582118>
>


-- 

*Ethan Ferrari*
Javascript Engineer

email *ethan@ethanferrari.com <ethan@ethanferrari.com>*
phone *309.657.0468*
code  *github.com/epferrari <http://github.com/epferrari>*